src/main/java/com/exasol/sql/rendering/ColumnsDefinitionRenderer.java

```diff
@@ -0,0 +1,123 @@
+package com.exasol.sql.rendering;
+
+import com.exasol.datatype.type.*;
+import com.exasol.datatype.type.Boolean;
+import com.exasol.sql.*;
+
+/**
+ * Renderer for columns definition.
+ */
+public class ColumnsDefinitionRenderer extends AbstractFragmentRenderer implements ColumnDefinitionVisitor {
+    /**
+     * Create a new instance of an {@link AbstractFragmentRenderer}-based class.
+     *
+     * @param config renderer configuration
+     */
+    public ColumnsDefinitionRenderer(final StringRendererConfig config) {
+        super(config);
+    }
+
+    @Override
+    public void visit(final ColumnsDefinition columnsDefinition) {
+        if (!columnsDefinition.getColumns().isEmpty()) {
+            appendSpace();
```

Comment:
That's better, but not what I mean. A renderer should not start a rendered fragment with a space. That should be added in the calling renderer.